sympy/polys/flint_powerseriesring.py

```diff
@@ -0,0 +1,255 @@
+
+    @property
+    def gen(self) -> ZZSeries:
+        return fmpz_poly([0, 1])
+
+    def pretty(self, s: ZZSeries) -> str:
+        """Pretty print a power series with improved formatting."""
+        if isinstance(s, fmpz_poly):
+            return series_from_list(s.coeffs(), None)
+
+        prec = int(s.repr().split('prec=')[1].split(')')[0].strip())
+        return series_from_list(s.coeffs(), prec)
+
+    def print(self, s: ZZSeries) -> str:
+        return self.pretty(s)
+
+    def equal(self, s1: ZZSeries, s2: ZZSeries) -> bool:
+        if isinstance(s1, fmpz_poly) and isinstance(s2, fmpz_poly):
+            return s1==s2
+        elif isinstance(s1, fmpz_series) and isinstance(s2, fmpz_series):
+            return s1._equal_repr(s2)
+
+        return False
+
+    def negative(self, s: ZZSeries) -> ZZSeries:
+        """Return the negative of a power series."""
+        return -s
+
+    def add(self, s1: ZZSeries, s2: ZZSeries) -> ZZSeries:
+        """Add two power series."""
+        with _global_cap(self.prec):
+            if isinstance(s1, fmpz_poly) and isinstance(s2, fmpz_poly):
+                poly = s1 + s2
+                if poly.degree() < self.prec:
+                    return poly
+                return fmpz_series(poly, prec=self.prec)
+
+            return s1 + s2
+
+    def subtract(self, s1: ZZSeries, s2: ZZSeries) -> ZZSeries:
+        """Subtract two power series."""
+        with _global_cap(self.prec):
+            if isinstance(s1, fmpz_poly) and isinstance(s2, fmpz_poly):
+                poly = s1 - s2
+                if poly.degree() < self.prec:
+                    return poly
+                return fmpz_series(poly, prec=self.prec)
+
+            return s1 - s2
+
+    def multiply(self, s1: ZZSeries, s2: ZZSeries) -> ZZSeries:
+        """Multiply two power series."""
+        with _global_cap(self.prec):
+            if isinstance(s1, fmpz_poly) and isinstance(s2, fmpz_poly):
+                poly = s1 * s2
+                if poly.degree() < self.prec:
+                    return poly
+                return fmpz_series(poly, prec=self.prec)
+
+            return s1 * s2
+
+    def multiply_ground(self, s1: ZZSeries, n: Any) -> ZZSeries:
+        """Multiply a power series by an integer or a polynomial."""
+        with _global_cap(self.prec):
+            if isinstance(s1, fmpz_poly):
+                poly = s1 * n
+                if poly.degree() < self.prec:
+                    return poly
+                return fmpz_series(poly, prec=self.prec)
+
+            return s1 * n
+
+    def trunc(self, s: ZZSeries, n: int) -> ZZSeries:
+        """Truncate a power series to the first n terms."""
+        if n < 0:
+            raise ValueError("Truncation precision must be non-negative")
+        with _global_cap(self.prec):
+            coeffs = s.coeffs()[:n]
+
+            if isinstance(s, fmpz_poly):
+                return fmpz_poly(coeffs)
+            return fmpz_series(coeffs, prec=n)
```

Comment:
Maybe this should return `fmpz_series` if some terms have been discarded from an `fmpz_poly`.

I'm not sure about calling this `trunc`. The Array API has trunc but it means something very different.

Reply:
The intention here is that in the end you could use this somewhat interchangeably with the array API like:
```python
import numpy as np
a = np.cos(np.ones(2, 2))

from sympy import series_ring, QQ
R = series_ring(QQ, 'x')
s = R.cos(R.one)
```
In so far as possible the functions of the same name should be equivalent so we should not have a function called `trunc` if it does not mean the same thing as `np.trunc`.